### PR TITLE
Carousel fixes

### PR DESCRIPTION
**Pull request by ds300**
Couple of small things

1. Screen dimensions updates seemed to be subject to a race condition on my iPad simulator at least. Calling `getScreenDimensions` synchronously with the `orientationChanged` callback was returning the pre-rotation values. So i moved the call onto the main thread right before the event is fired across the bridge.
2. Images whose maximum resolution is smaller than the viewport were being blown up to fill the screen in the full screen carousel. On iPad this could look *really* bad so I prevented it from happening.

#trivial
#skip_new_tests

### CODE REVIEW

File: Pod/Classes/Core/ARDynamicScreenDimensions.m [hunk 1]
```diff
@@ -64,17 +64,18 @@ + (BOOL)requiresMainQueueSetup
...
+  dispatch_async(dispatch_get_main_queue(), ^{
+    __strong ARDynamicScreenDimensions *sself = wself;
+    if (!sself) {
+      return;
+    }
+    sself.currentDimensions = getScreenDimensions();
+    if (sself.isBeingObserved) {
+      [sself sendEventWithName:@"change" body:sself.currentDimensions];
+    }
+  });
```
Comment (alloy) [inline Pod/Classes/Core/ARDynamicScreenDimensions.m]: The documentation for this notification indicates that the `NSNotification` object actually holds the value, which seems like a less brittle approach and perhaps less blocking of the main thread https://developer.apple.com/documentation/uikit/uideviceorientationdidchangenotification?language=objc

Comment (alloy) [reply Pod/Classes/Core/ARDynamicScreenDimensions.m]: Or did you mean the values in the size struct?

Comment (ds300) [reply Pod/Classes/Core/ARDynamicScreenDimensions.m]: yeah the sizes and safe area insets were wrong too.

Comment (alloy) [reply Pod/Classes/Core/ARDynamicScreenDimensions.m]: Gotcha

**Pull request merged**